src/basic/env-util.c

```diff
@@ -452,6 +452,18 @@ char *strv_env_get(char **l, const char *name) {
         return strv_env_get_n(l, name, strlen(name), 0);
 }
 
+char *strv_env_pairs_get(char **l, const char *name) {
+        char **i;
+
+        assert(name);
+
+        STRV_FOREACH_BACKWARDS(i, l)
+                if (streq(*i, name) && i + 1)
+                        return *(i + 1);
```

Comment:
hmm, the input string arrays is supposed to have pairs of entries, right? i.e. xn, xv, yn, yv, zn, zv, …? here you iterate through every individual entry backwards, i.e. you might find xv match your search key and then end up return yn, but that' be borked.

just iterate fowards, with STRV_FOREACH_PAIR() and store a way a pointer to the last value that had a matching name. it's going to be nicer performance-wise, too, since that way you only iterate through the list once instead of twice

Reply:
ah good catch - fixed